src/Microsoft.ML.AutoML/AutoMLExperiment/TrialSettingsProposer/PipelineProposer.cs

```diff
@@ -0,0 +1,270 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.ML.AutoML.CodeGen;
+using Newtonsoft.Json;
+using static Microsoft.ML.AutoML.AutoMLExperiment;
+
+namespace Microsoft.ML.AutoML
+{
+    /// <summary>
+    /// propose sweepable estimator pipeline from a group of candidates using eci in flaml (https://arxiv.org/abs/1911.04706)
+    /// </summary>
+    internal class PipelineProposer : ISavableProposer
+    {
+        private readonly Dictionary<EstimatorType, double> _estimatorCost;
+        private Dictionary<string, double> _learnerInitialCost;
+
+        // total time spent on last best error for each learner.
+        private Dictionary<string, double> _k1;
+
+        // total time spent on second last best error for each learner.
+        private Dictionary<string, double> _k2;
+
+        // last best error for each learner.
+        private Dictionary<string, double> _e1;
+
+        // second last best error for each learner.
+        private Dictionary<string, double> _e2;
+
+        // flaml ECI
+        private Dictionary<string, double> _eci;
+        private double _globalBestError;
+
+        private readonly Random _rand;
+        private MultiModelPipeline _multiModelPipeline;
+
+        public PipelineProposer(AutoMLExperimentSettings settings)
+        {
+            this._estimatorCost = new Dictionary<EstimatorType, double>()
+            {
+                { EstimatorType.LightGbmRegression, 0.788 },
+                { EstimatorType.FastTreeRegression, 0.382 },
```

Comment:
Where are the numbers in this list coming from?

Reply:
It's from a internal benchmark that we ran for ModelBuilder.AutoMLService, essentially the smaller the number, the less time it spent on its init parameter, which make it easier to be picked when experiment starts

Reply:
Could you add a comment that says taht? Even if you don't say you got it from ModelBuilder, can you at least say that the smaller the number the less time taken on init? Just so we can remember in the future.

Reply:
Yeah absolutely